DNN Platform/DotNetNuke.Web.Client.ResourceManager/ClientResourceController.cs

```diff
@@ -0,0 +1,217 @@
+    using DotNetNuke.Abstractions.ClientResources;
+
+    /// <inheritdoc />
+    public class ClientResourceController : IClientResourceController
+    {
+        private readonly IHostSettings hostSettings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClientResourceController"/> class.
+        /// </summary>
+        /// <param name="hostSettings">The host settings.</param>
+        public ClientResourceController(IHostSettings hostSettings)
+        {
+            this.hostSettings = hostSettings;
+            this.RegisterPathNameAlias("SharedScripts", "~/Resources/Shared/Scripts/");
+        }
+
+        private List<IFontResource> Fonts { get; set; } = new List<IFontResource>();
+
+        private List<IScriptResource> Scripts { get; set; } = new List<IScriptResource>();
+
+        private List<IStylesheetResource> Stylesheets { get; set; } = new List<IStylesheetResource>();
+
+        private Dictionary<string, string> PathNameAliases { get; set; } = new Dictionary<string, string>();
+
+        private List<string> FontsToExclude { get; set; } = new List<string>();
+
+        private List<string> ScriptsToExclude { get; set; } = new List<string>();
+
+        private List<string> StylesheetsToExclude { get; set; } = new List<string>();
+
+        /// <inheritdoc />
+        public void AddFont(IFontResource font)
+        {
+            this.Fonts = this.AddResource(this.Fonts, font);
+        }
+
+        /// <inheritdoc />
+        public void AddScript(IScriptResource script)
+        {
+            this.Scripts = this.AddResource(this.Scripts, script);
+        }
+
+        /// <inheritdoc />
+        public void AddStylesheet(IStylesheetResource stylesheet)
+        {
+            this.Stylesheets = this.AddResource(this.Stylesheets, stylesheet);
+        }
+
+        /// <inheritdoc />
+        public IFontResource CreateFont()
+        {
+            return new Models.FontResource(this);
+        }
+
+        /// <inheritdoc />
+        public IScriptResource CreateScript()
+        {
+            return new Models.ScriptResource(this);
+        }
+
+        /// <inheritdoc />
+        public IStylesheetResource CreateStylesheet()
+        {
+            return new Models.StylesheetResource(this);
+        }
+
+        /// <inheritdoc />
+        public void RegisterPathNameAlias(string pathNameAlias, string resolvedPath)
+        {
+            this.PathNameAliases[pathNameAlias.ToLowerInvariant()] = resolvedPath;
+        }
+
+        /// <inheritdoc />
+        public void RemoveFontByName(string fontName)
+        {
+            this.FontsToExclude.Add(fontName.ToLowerInvariant());
+        }
+
+        /// <inheritdoc />
+        public void RemoveFontByPath(string fontPath, string pathNameAlias)
+        {
+            var fullPath = this.ResolvePath(fontPath, pathNameAlias).ToLowerInvariant();
+            this.FontsToExclude.Add(fullPath.ToLowerInvariant());
```

Comment:
We don't need to call `ToLowerInvariant` twice (or at all if we change it to a HashSet with a case-insensitive comparer).
```suggestion
            var fullPath = this.ResolvePath(fontPath, pathNameAlias);
            this.FontsToExclude.Add(fullPath.ToLowerInvariant());
```